src/mqtt_client/mod.rs

```diff
@@ -339,58 +409,82 @@ where
     {
         let settings = &mut self.settings;
         let mqtt = &mut self.mqtt;
-        let prefix = self.settings_prefix.as_str();
-
-        let mut response_topic: String<MAX_TOPIC_LENGTH> = String::from(self.prefix.as_str());
-        response_topic.push_str("/log").unwrap();
-        let default_response_topic = response_topic.as_str();
+        let prefix = self.prefix.as_str();
+        let listing_state = &mut self.listing_state;
 
         let mut updated = false;
         match mqtt.poll(|client, topic, message, properties| {
-            let path = match topic.strip_prefix(prefix) {
-                // For paths, we do not want to include the leading slash.
-                Some(path) => {
-                    if !path.is_empty() {
-                        &path[1..]
-                    } else {
-                        path
-                    }
+            let Some(path) = topic.strip_prefix(prefix) else {
+                info!("Unexpected MQTT topic: {}", topic);
+                return;
+            };
+
+            let Ok(command) = Command::from_str(path) else {
+                info!("Unknown Miniconf command: {path}");
+                return;
+            };
+
+            let mut data = [0u8; MESSAGE_SIZE];
+            let message = match command {
+                Command::Query => {
+                    let path = core::str::from_utf8(message).unwrap_or("");
+
+                    // Get a specific settings value with the provided path.
+                    settings
+                        .get(path, &mut data)
+                        .map(|len| &data[..len])
+                        .unwrap_or(b"Invalid path")
                 }
-                None => {
-                    info!("Unexpected MQTT topic: {}", topic);
+
+                Command::List => {
+                    let Some(response_topic) = properties.into_iter().find_map(|p| {
+                        if let Ok(minimq::Property::ResponseTopic(topic)) = p {
```

Comment:
Also carry over correlation data into the response(s)
I think we already do this for the set response and/or in minireq.

Reply:
This is an interesting problem, as we will have to cache the `CorrelationData` outside of the minimq handler. We should already have enough buffer space for this (because we reserve the max MQTT message size), but I'll have to look at how we can cache both the response topic and CD. It's definitely possible, but may require minimq updates

Reply:
Can't you just save and then copy/clone the empty-payload `Publication` and take things from there?